eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -46,6 +46,40 @@ case class RemoteCommit(index: Long, spec: CommitmentSpec, txid: ByteVector32, r
 case class WaitingForRevocation(nextRemoteCommit: RemoteCommit, sent: CommitSig, sentAfterLocalCommitIndex: Long, reSignAsap: Boolean = false)
 // @formatter:on
 
+trait ChannelCommitments {
+  def getOutgoingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc]
+
+  def getIncomingHtlcCrossSigned(htlcId: Long): Option[UpdateAddHtlc]
+
+  def timedOutOutgoingHtlcs(blockheight: Long): Set[UpdateAddHtlc]
+
+  def localNodeId: PublicKey
+
+  def remoteNodeId: PublicKey
+
+  val availableBalanceForReceive: MilliSatoshi
+
+  val availableBalanceForSend: MilliSatoshi
+
+  val originChannels: Map[Long, Origin]
+
+  val channelId: ByteVector32
+
+  val announceChannel: Boolean
+
+  def failHtlc(nodeSecret: PrivateKey, cmd: CMD_FAIL_HTLC, add: UpdateAddHtlc): Try[UpdateFailHtlc] = {
+    Sphinx.PaymentPacket.peel(nodeSecret, add.paymentHash, add.onionRoutingPacket) match {
+      case Right(Sphinx.DecryptedPacket(_, _, sharedSecret)) =>
+        val reason = cmd.reason match {
+          case Left(forwarded) => Sphinx.FailurePacket.wrap(forwarded, sharedSecret)
+          case Right(failure) => Sphinx.FailurePacket.create(sharedSecret, failure)
+        }
+        Success(UpdateFailHtlc(channelId, cmd.id, reason))
+      case Left(_) => Failure(CannotExtractSharedSecret(channelId, add))
+    }
+  }
```

Comment:
I don't think this makes sense as a method inside the trait.
It's probably better to make this a static helper function inside the `Commitments` object, WDYT?

Reply:
I did this because this chunk can be directly reused in HC while `Commitments` object is related to specifically normal channel commitments (?). Maybe `ChannelCommitments` object?

Reply:
Actually a good place to put that helper function could be in `PaymentPacket.scala`, in the `OutgoingPacket` object.
I would also rename it `buildHtlcFailure` to be more consistent with that file's functions.